tests/conftest.py

```diff
@@ -0,0 +1,40 @@
+import glob
+import json
+import os
+
+import pytest
+from nomad.utils import structlogging
+
+
+@pytest.fixture(scope='function')
+def capture_from_logger(caplog):
```

Comment:
Do we want this to be a fixture or rather in `utils`? I see some utility on having a method which returns True or False (together with the message) if there are ERROR messages in the logger.

I will even go one step further, and say that we can have a function which returns True or False depending on whether "ERROR" and "WARNING", depending on an input passed. Something like:
```python
def capture_msg_from_logger(caplog, levelname):
    if levelname not in ['ERROR', 'WARNING']:
         return None, None  # this is done to recognize if levelname is neither ERROR nor WARNING
    caplog.handler.formatter = structlogging.ConsoleFormatter()
    yield caplog
    for record in caplog.get_records(when='call'):
        if record.levelname in levelname:
            try:
                msg = structlogging.ConsoleFormatter.serialize(json.loads(record.msg))
            except Exception:
                msg = record.msg
            return True, msg
    return False, None
```

If you decide to go this way, please, make sure to add typing + add error handler to check that the `caplog` is what you expect and that it has handler and handler.formatter defined.

Reply:
@JosePizarro3 Hmm. There could be some benefits to also get warnings, but we can also have it as another fixture, which simply prints out the warnings.

I like the idea of having fixtures for the pytest because it is well integrated with the tests: we don't need to import any additional functions from utils or put conditionals for executing these functions in the test_functions.

Also, we don't have to check caplog because it is a standard fixture from pytest that has the handler property.

Reply:
Ok good. My point was related with the error and testing from below. Hence, if you don't go with the idea I suggest, then you can keep this as a fixture. Otherwise, if you accept my suggestion on testing, I keep suggesting of moving this to utils (because it is useful, but you don't need it for testing 🙂 )

Reply:
I really think this should be a part of testing/debugging. Exploring the errors in the logs isn't something that plugin src should contain, unless it's useful for some plugin functionality